### PR TITLE
Add descriptions for Search API

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Adds descriptions for the Search API. Source for the descriptions: https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html (and pages linked from it)

**Does not include aggs and suggesters.**

### CODE REVIEW

**Comment by abdonpijpelink** [comment]
I'm not sure what to do about the validation error (if anything). Running `make validate api=search stack-version=8.8.0` locally just returns:

```
✖ 
    at run (/Users/abdon/projects/github/elasticsearch-specification/compiler/run-validations.js:175:12)
    exit code: 1
make: *** [validate] Error 1
```

File: specification/_global/search/SearchRequest.ts [hunk 1]
```diff
@@ -51,69 +51,290 @@ import { RankContainer } from '@_types/Rank'
...
+     * When possible, let Elasticsearch perform early termination automatically.
+     * Avoid specifying this parameter for requests that target data streams with backing indices across multiple data tiers.
+     * If set to `0` (default), the query does not terminate early.
+     * @server_default 0
+     */
     terminate_after?: long
+    /**
+     * Specifies the period of time to wait for a response from each shard.
+     * If no response is received before the timeout expires, the request fails and returns an error.
+     * @server_default no timeout
```
Comment (lcawl) [inline specification/_global/search/SearchRequest.ts:246]: Is "no timeout" actually a valid value for this property? I'd have assumed it was a numeric value:
```suggestion
     * @server_default 0
```
If we're not sure, I think it's better to omit it.

**Comment by lcawl** [comment]
> I'm not sure what to do about the validation error (if anything). 

I think the validation messages can be ignored in the context of adding descriptions. My interpretation of them is that they're indicating there are some request and response properties missing from the spec, but that's not what we're addressing with these PRs, unless we happen to notice an obvious omission.

**Review by lcawl — APPROVED** [review]
One suggestion, otherwise LGTM

**Pull request merged**